mne/decoding/search_light.py

```diff
@@ -0,0 +1,491 @@
+# Author: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from .mixin import TransformerMixin
+from .base import BaseEstimator  # XXX reorganize sklearn objects.
+from ..parallel import parallel_func
+
+
+class SearchLight(BaseEstimator, TransformerMixin):
+    """Search Light.
+
+    Fit, predict and score a series of models to each subset of the dataset
+    along the third dimension.
+
+    Parameters
+    ----------
+    base_estimator : object
+        The base estimator to iteratively fit on a subset of the dataset.
+    n_jobs : int, optional (default=1)
+        The number of jobs to run in parallel for both `fit` and `predict`.
+        If -1, then the number of jobs is set to the number of cores.
+    """
+    def __init__(self, base_estimator, n_jobs=1):
+        self.base_estimator = base_estimator
```

Comment:
I think I should add a test at init to check that the estimator is an indeed an estimator and not 'foo'. 

This check function has been added in 3447, so might as well wait for it to be merged